package/lib/src/beamer_router_delegate.dart

```diff
@@ -451,6 +452,20 @@ class BeamerRouterDelegate<T extends BeamState> extends RouterDelegate<Uri>
       _currentLocation.removeListener(_notify);
       _pushHistory(notFoundRedirect!);
     }
+
+    if (kIsWeb) {
```

Comment:
I think we should move this

```dart
if (_currentPages.isEmpty) {
  _currentLocation =
      NotFound(path: _currentLocation.state.uri.toString());
} else if (kIsWeb) {
  // HERE
}
```

This way you will have the right `context` (if someone is overriding `BeamLocation.builder`), you don't have to worry about empty pages list and you will also have `_currentPages` to use instead of calling `pagesBuilder`